File: C,C++/p10935.cpp
#include <iostream>
#include <deque>
#include <vector>
#include <iterator>

using namespace std;

int main () {
	while (1) {
		int banyakN;
		cin >> banyakN;
		if (banyakN == 0) break;
		
		deque<int> tumpukanDeck;
		vector<int> yangDibuang;
		for (int i = 1; i <= banyakN; i++) {
			tumpukanDeck.push_back(i);
		}
		
		while (tumpukanDeck.size() != 1) {
			int temp = tumpukanDeck.front();
			tumpukanDeck.pop_front();
			yangDibuang.push_back(temp);
			temp = tumpukanDeck.front();
			tumpukanDeck.pop_front();
			tumpukanDeck.push_back(temp);
		}
		
		if (yangDibuang.size() != 0) cout << "Discarded cards: ";
		else cout << "Discarded cards:";
		
		int tempSize = yangDibuang.size();
		int temp = 1;
		
		for (auto i = yangDibuang.begin(); i != yangDibuang.end(); ++i) {
			if (temp != tempSize) {
				cout << *i << ", ";
			} else {
				cout << *i;
			}
			temp++;
		}
		
		cout<<endl;
		
		cout << "Remaining card: ";
		cout << tumpukanDeck.front() << endl;
	}
	return 0;
}

File: Java/p10935.java
import java.util.Deque;
import java.util.Queue;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.LinkedList;
import java.util.Iterator;

public class p10935 {
	public static void main (String [] args) throws IOException {
		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
		while (true) {
			int num = Integer.parseInt(input.readLine());
			if (num == 0) break;
			Queue<Integer> yangDibuang = new LinkedList<>();
			Deque<Integer> tumpukanDeck = new LinkedList<>();
			
			for (int i = 1; i <=num; i++) {
				tumpukanDeck.addLast(i);
			}
			
			while (tumpukanDeck.size() != 1) {
				yangDibuang.offer(tumpukanDeck.pop());
				tumpukanDeck.addLast(tumpukanDeck.pop());
			}
			
			if (yangDibuang.size() == 0) System.out.println("Discarded cards:");
			else System.out.print("Discarded cards: ");
			
			int tempSize = yangDibuang.size();
			int temp = 1;
			
			Iterator it = yangDibuang.iterator();
			while (it.hasNext()) {
				if (temp != tempSize) {
					System.out.print(it.next() + ", ");
				} else {
					System.out.println(it.next());
				}
				temp++;
			}
			
			System.out.println("Remaining card: " + tumpukanDeck.pop());
		}
	}
}